app/models/concerns/orchestration/dhcp.rb

```diff
@@ -102,7 +102,7 @@ def dhcp_attrs
     }
 
     if provision?
-      dhcp_attr.merge!({:filename => operatingsystem.boot_filename(self), :nextServer => boot_server})
+      dhcp_attr.merge!({:filename => operatingsystem.boot_filename(self.host), :nextServer => boot_server})
```

Comment:
It's NIC only according to my records.

```
Searching 2878 files for "Orchestration::DHCP"

/home/lzap/work/foreman/app/models/concerns/orchestration/dhcp.rb:
    1: module Orchestration::DHCP
    2    extend ActiveSupport::Concern
    3    include Orchestration::Common

/home/lzap/work/foreman/app/models/nic/managed.rb:
    2    class Managed < Interface
    3      include Orchestration
    4:     include Orchestration::DHCP
    5      include Orchestration::DNS
    6      include Orchestration::TFTP

2 matches across 2 files
```

But the `boot_filename` param name is called `host` so I assumed this was wrong. Some OSes use that to get MAC address (work on both objects) and Solaris calls `model`. In all cases these are aliases to host, I see no reason to send NIC there.